language/diem-framework/compiled/Cargo.toml

```diff
@@ -11,15 +11,13 @@ publish = false
 
 [dependencies]
 anyhow = "1.0.38"
-bytecode-verifier = { path = "../../bytecode-verifier", version = "0.1.0" }
+bcs = "0.1.2"
 diem-crypto = { path = "../../../crypto/crypto", version = "0.1.0" }
 diem-types = { path = "../../../types", version = "0.1.0" }
 diem-workspace-hack = { path = "../../../common/workspace-hack", version = "0.1.0" }
-bcs = "0.1.2"
-vm = { path = "../../vm", version = "0.1.0" }
-diem-framework = { path = "../../diem-framework",  version = "0.1.0" }
```

Comment:
This is the key change I would like to address, dropping the dependency from `compiled-stdlib` to `stdlib` (now `diem-framework`.

The reason is because of the dependency chain:

- ir-compiler
- compiled-stdlib
- diem-framework
- move-lang
- move-prover
- <all-other-kinds-of-stuff>

Therefore, whenever I change the IR-compiler, I need to make sure that all other components compile as well... Which becomes very hard when working on the lifting Vector functions into bytecode instructions PR.

Removing this dependency here will ease the situation a bit. Plus, to me it also makes sense logically as why loading bytes from the disk should require all other components in the repo...

Reply:
No, it is the `compiler/Cargo.toml`

Reply:
Oh I see. I confused myself.

However I'm not sure your suggested change will solve your problem. Note that `language/compiler` and `language/compiler/ir-to-bytecode` are different crates. 
![graphviz (5)](https://user-images.githubusercontent.com/10379359/108906682-00b14200-75d6-11eb-84e0-a1d64f971e0f.png)
When you said you changed the IR compiler, you were probably referring to `ir-to-bytecode`, which is used by the bunch of stuff you mentioned. If I'm understanding correctly, removing the dependency from `compiled-stdlib` to `diem-framework` will only free `language/compiler` from having to depend on a bunch of things but not reduce time of rebuilds triggered by a change in `ir-to-bytecode`.

With that being said I'm not against this change at all!

Reply:
Sorry, it is my bad that I did not specify clearly what the IR compiler is. Thanks for the clarification!

> removing the dependency from compiled-stdlib to diem-framework will only free language/compiler from having to depend on a bunch of things but not reduce time of rebuilds triggered by a change in ir-to-bytecode.

Totally!